src/home/fetch-github/fetch-issues-preview.ts

```diff
@@ -17,8 +20,23 @@ export async function fetchIssuePreviews(): Promise<TaskNoFull[]> {
     if (403 === error.status) {
       console.error(`GitHub API rate limit exceeded.`);
       if (taskManager.getTasks().length == 0) {
-        // automatically login if there are no issues loaded
-        automaticLogin(error);
+        if (!user || user === null) {
+          // only show rate limit modal if there are no issues loaded and not logged in
+          rateLimitModal(error);
+        } else {
+          // otherwise we have a user and no issues loaded
+          // this happens processing the auth token it seems
+          // as it happens on auth callback and local auth token
+          await gitHubLoginButtonHandler().catch((error) => {
+            console.error(error);
+          });
+        }
+      } else if (user && user !== null) {
+        // Tasks loaded and logged in
+        rateLimitModal(error, `You have been rate limited. Please try again at `);
```

Comment:
I think it makes more sense to directly pass in the full string here, including the timestamp, instead of handling the timestamp generating in the `rateLimitModal` function.